### PR TITLE
API: Remove NINF and PINF usages

### PR DESCRIPTION
**Pull request by mtsokol**
Hi!
Due to NumPy's main namespace being changed in https://github.com/numpy/numpy/pull/24357, here I update warning imports.

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -100,9 +100,6 @@ def canonicalize_shape(shape: Any, context: str="") -> core.Shape:
 e = np.e
 euler_gamma = np.euler_gamma
 inf = np.inf
-NINF = np.NINF
```
Comment (hawkinsp) [inline jax/_src/numpy/lax_numpy.py:103]: Let's leave these in for now with a `# TODO: reason` comment to remove them.

You could change them to:
```
NINF = -np.inf
```
to fix the NumPy issue.

The reason is that these are reexporting NumPy's constants as part of JAX's API, and to change JAX's API we should follow a deprecation process.

File: jax/numpy/__init__.py [hunk 2]
```diff
@@ -22,9 +22,6 @@
 
 from jax._src.numpy.lax_numpy import (
     ComplexWarning as ComplexWarning,
```
Comment (hawkinsp) [inline jax/numpy/__init__.py:24]: Leave these in, and add a TODO.

**Review by hawkinsp — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: jax/_src/numpy/lax_numpy.py [hunk 3]
```diff
@@ -100,9 +100,6 @@ def canonicalize_shape(shape: Any, context: str="") -> core.Shape:
...
-PZERO = np.PZERO
```
Comment (hawkinsp) [inline jax/_src/numpy/lax_numpy.py:104]: Out of curiosity, what should users replace `PZERO` and `NZERO` with? `0.0` and `-0.0`?

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (mtsokol) [reply jax/_src/numpy/lax_numpy.py:103]: Sure! I moved them back.

File: jax/_src/numpy/lax_numpy.py (hunk 3 above)
Comment (mtsokol) [reply jax/_src/numpy/lax_numpy.py:104]: Yes, in numpy's Python API, these should be used.

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (hawkinsp) [reply jax/_src/numpy/lax_numpy.py:103]: It's fine to change these to be *defined* in terms of the new NumPy names, though (`-np.inf`, etc.). That would avoid any warnings/errors from upstream NumPy.

File: jax/numpy/__init__.py (hunk 2 above)
Comment (mtsokol) [reply jax/numpy/__init__.py:24]: Sure! Added TODO with numpy's version that will ship this change.

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (hawkinsp) [reply jax/_src/numpy/lax_numpy.py:103]: Ah, you did that. Perfect.

File: jax/_src/numpy/lax_numpy.py (hunk 3 above)
Comment (hawkinsp) [reply jax/_src/numpy/lax_numpy.py:104]: It'd be helpful if there were a table somewhere of replacements (e.g., in the NumPy release notes?)

**Review by hawkinsp — APPROVED** [review]

File: jax/_src/numpy/lax_numpy.py (hunk 3 above)
Comment (mtsokol) [reply jax/_src/numpy/lax_numpy.py:104]: Currently, it's in an issus, as this list still evolves: https://github.com/numpy/numpy/issues/24306#issuecomment-1663564574 (when you open "details" in this comment you will see current state of main namespace. I will decorate each "removed" with recommended replacement).

Yes, release notes will contain all changes that are being introduced right now.

**Pull request merged**